components/micro-gateway-core/src/main/ballerina/gateway/executiontime_util.bal

```diff
@@ -0,0 +1,62 @@
+import ballerina/log;
+import ballerina/io;
+import ballerina/http;
+
+
+function getExecutionTimePayload(ExecutionTimeDTO executionTimeDTO) returns string {
+    string output = executionTimeDTO.apiName + OBJ + executionTimeDTO.apiVersion + OBJ + executionTimeDTO.tenantDomain +
+        OBJ + executionTimeDTO.provider + OBJ + executionTimeDTO.apiResponseTime + OBJ + executionTimeDTO.context + OBJ
+        + executionTimeDTO.securityLatency + OBJ + executionTimeDTO.throttlingLatency + OBJ + executionTimeDTO.
+        requestMediationLatency + OBJ +
+        executionTimeDTO.responseMediationLatency + OBJ + executionTimeDTO.backEndLatency + OBJ + executionTimeDTO.
+        otherLatency + OBJ + executionTimeDTO.eventTime;
+
+    return output;
+}
+
+function getMetaDataForExecutionTimeDTO(ExecutionTimeDTO dto) returns string {
+    return "{\\\"keyType\\\":\"" + dto.keyType + "\",\\\"correlationID\\\":\"" + dto.correleationID + "\"}";
```

Comment:
Can't we create a Json payload and convert to string in Ballerina?

Reply:
Fixed in  https://github.com/wso2/product-microgateway/pull/148